airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -47,14 +47,14 @@ def __init__(self, *args, aws_conn_id: str = "aws_conn_id", hook_params: dict |
         if hook_params is not None:
             warnings.warn(
                 "The parameter hook_params is deprecated and will be removed. "
+                "Note that it is also incompatible with deferrable mode. "
                 "If you were using it, please get in touch either on airflow slack, "
                 "or by opening a github issue on the project. "
                 "You can mention https://github.com/apache/airflow/pull/32352",
                 AirflowProviderDeprecationWarning,
                 stacklevel=3,  # 2 is in the operator's init, 3 is in the user code creating the operator
             )
-        self.hook_params = hook_params or {}
```

Comment:
What happens if someone wants to pass something like `region_name` to the hook? This is a fairly common use case, and there is no reason to exclude that from being used in deferrable operators.

Reply:
There is a bunch of operators that don't allow passing a region at all, like the sagemaker operators, DMS, etc.

If we want users to be able to pass a region, I think we should add that as an explicit parameter.

Reply:
pushed a change to that effect.

Reply:
This is something we might want to think about actually. In which case a user might want to specify another region? Regions are set as part of connections. That means, a user set `regionA` in its connection, hence targeting this regions for all AWS calls but would want to target another region for a specific operator? I think that can be done by using another connection then (and then using the param `aws_conn_id`). WDYT?

Reply:
I don't exactly know how that works in practice... I suppose you'd have a "main" region where you do the heavy lifting (sagemaker stuff for instance), but you'd have DBs in various regions, and you'd like to be able to hit those without having to change too much stuff ?
Are you saying that the region param should be obsoleted everywhere in the AWS provider in favor of several connection ids ?

Reply:
It feels duplicate to me ... But that's my developer opinion, I can understand, as a user, just passing a region as parameter is easier than creating a new connection. Maybe a question for @shubham22

Reply:
>There is a bunch of operators that don't allow passing a region at all, like the sagemaker operators, DMS, etc.

I agree with you here, but in the interest of keeping things backwards compatible, if a particular operator allows passing region config, then we should continue that or if we want to stop supporting that, then we would need to go through the whole deprecating process. 
At this point, I don't know how you want to proceed though :p

Reply:
> just passing a region as parameter is easier than creating a new connection

I agree with this. I haven't had direct conversation on this with customers, but I would assume this is commonly done. Yes, many operators do not support it today and may be some users might request for them in the future. In any case, we shouldn't take it away if it is supported until and unless we think it is causing some other problems.